usecases/objects/batch_references_add.go

```diff
@@ -117,6 +127,110 @@ func (b *BatchManager) validateReference(wg *sync.WaitGroup, ref *models.BatchRe
 	}
 }
 
+func validateReferenceMultiTenancy(ctx context.Context,
+	principal *models.Principal, schemaManager schemaManager,
+	repo VectorRepo, source *crossref.RefSource, target *crossref.Ref,
+	tenantKey string,
+) error {
+	if source == nil || target == nil {
+		return fmt.Errorf("can't validate multi-tenancy for nil refs")
+	}
+
+	sourceClass, targetClass, err := getReferenceClasses(
+		ctx, principal, schemaManager, source.Class.String(), target.Class)
+	if err != nil {
+		return err
+	}
+
+	// if both classes have MT enabled, no cross-tenant references can be made
+	if sourceClass.MultiTenancyConfig != nil && targetClass.MultiTenancyConfig != nil {
+		if sourceClass.MultiTenancyConfig.TenantKey != targetClass.MultiTenancyConfig.TenantKey {
+			return fmt.Errorf("invalid reference: source class %q tenant key %q "+
+				"is different than target class %q tenant key %q",
+				sourceClass.Class, sourceClass.MultiTenancyConfig.TenantKey,
+				targetClass.Class, targetClass.MultiTenancyConfig.TenantKey)
+		}
+
+		err = validateTenantRefObjects(ctx, repo,
+			sourceClass, targetClass, source, target, tenantKey)
+		if err != nil {
+			return err
+		}
+	}
+
+	// the other way around is ok, a MT-enabled class may reference a
+	// non-MT-enabled class.
+	if sourceClass.MultiTenancyConfig == nil && targetClass.MultiTenancyConfig != nil {
```

Comment:
I think it might cheaper for this check to be done before line 146

Reply:
Interesting, how so? This is essentially an `else if` to the `if` on line 146

Reply:
Yes you're right.

Even if hte conditions on lines 146 and 164 are disjoint, reodering them won't be more efficient